src/libraries/System.Private.CoreLib/src/System/Threading/CancellationTokenSource.cs

```diff
@@ -446,7 +446,9 @@ private void CancelAfter(uint millisecondsDelay)
                 }
             }
 
-            timer.Change(TimeSpan.FromMilliseconds(millisecondsDelay), Timeout.InfiniteTimeSpan);
+            timer.Change(
+                millisecondsDelay == Timeout.UnsignedInfinite ? Timeout.InfiniteTimeSpan : TimeSpan.FromMilliseconds(millisecondsDelay),
```

Comment:
I wonder if it's better to accept `TimeSpan` instead of `uint` here. That way you wouldn't even need this check.

Reply:
Someome could look at refactoring it separately.